### PR TITLE
Fix two minor issues. Closes #3812 and #3770

### LINKED ISSUE #3812: .changed() detects value as changed when set from null to null.
If the value of an instance was `null` and is `.set()` to `null`, Sequelize detects the value as changed.
Using sequelize 3.1.1.

### PR DESCRIPTION
**Pull request by BridgeAR**

### CODE REVIEW

File: lib/instance.js [hunk 1]
```diff
@@ -321,8 +321,7 @@ module.exports = (function() {
               value = (value === 1) ? true : (value === 0) ? false : value;
             }
           }
-
-          if (!options.raw && (primitives.indexOf(typeof value) === -1 || value !== originalValue)) {
+          if (!options.raw && (primitives.indexOf(typeof value) === -1 && value !== null || value !== originalValue)) {
```
Comment (mickhansen) [inline lib/instance.js]: Can you wrap the `&&` in additional parens just for clarity?

File: test/unit/instance/changed.test.js [hunk 2]
```diff
@@ -31,14 +31,17 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
...
       }, {
         isNewRecord: false,
         raw: true
       });
 
       user.set('name', 'a');
+      user.set('meta', null);
       expect(user.changed('name')).to.equal(false);
+      expect(user.changed('meta')).to.equal(false);
     });
```
Comment (mickhansen) [inline test/unit/instance/changed.test.js]: We might want a test for checking that `null` does return in changed if it was a non-null value before.

File: lib/instance.js (hunk 1 above)
Comment (BridgeAR) [inline lib/instance.js]: I'd say it's easier to read it without the extra parentheses but it's not really important...

**Comment by BridgeAR** [comment]
Comments incorporated

**Pull request merged**